py/picca/delta_extraction/astronomical_objects/forest.py

```diff
@@ -440,63 +361,32 @@ def rebin(self):
-
-            w1 = (self.log_lambda >= Forest.log_lambda_grid[0] - half_delta_log_lambda)
-            w1 = w1 & (self.log_lambda < Forest.log_lambda_grid[-1] + half_delta_log_lambda)
-            w1 = w1 & (self.log_lambda - np.log10(1. + self.z) >=
-                       Forest.log_lambda_rest_frame_grid[0] - half_delta_log_lambda_rest_frame)
-            w1 = w1 & (self.log_lambda - np.log10(1. + self.z) <
-                       Forest.log_lambda_rest_frame_grid[-1] + half_delta_log_lambda_rest_frame)
-            w1 = w1 & (self.ivar > 0.)
-            if w1.sum() == 0:
-                self.log_lambda = np.array([])
-                self.flux = np.array([])
-                self.ivar = np.array([])
-                self.transmission_correction = np.array([])
-                return [], [], [], [], []
-            self.log_lambda = self.log_lambda[w1]
-            self.flux = self.flux[w1]
-            self.ivar = self.ivar[w1]
-            self.transmission_correction = self.transmission_correction[w1]
-
-            bins = find_bins(self.log_lambda, Forest.log_lambda_grid)
-            self.log_lambda = Forest.log_lambda_grid[0] + bins * delta_log_lambda
-
-        elif Forest.wave_solution == "lin":
-            delta_lambda = Forest.lambda_grid[1] - Forest.lambda_grid[0]
-            half_delta_lambda = delta_lambda / 2.
-
-            half_delta_lambda_rest_frame = (Forest.lambda_rest_frame_grid[1] -
-                                            Forest.lambda_rest_frame_grid[0]) / 2.
-            w1 = (self.lambda_ >= Forest.lambda_grid[0] - half_delta_lambda)
-            w1 = w1 & (self.lambda_ < Forest.lambda_grid[-1] + half_delta_lambda)
-            w1 = w1 & (self.lambda_ / (1. + self.z) >= Forest.lambda_rest_frame_grid[0] - half_delta_lambda_rest_frame)
-            w1 = w1 & (self.lambda_ / (1. + self.z) < Forest.lambda_rest_frame_grid[-1] + half_delta_lambda_rest_frame)
-            w1 = w1 & (self.ivar > 0.)
-            if w1.sum() == 0:
-                self.lambda_ = np.array([])
-                self.flux = np.array([])
-                self.ivar = np.array([])
-                self.transmission_correction = np.array([])
-                return [], [], [], [], []
-            self.lambda_ = self.lambda_[w1]
-            self.flux = self.flux[w1]
-            self.ivar = self.ivar[w1]
-            self.transmission_correction = self.transmission_correction[w1]
-
-            bins = find_bins(self.lambda_, Forest.lambda_grid)
-            self.lambda_ = Forest.lambda_grid[0] + bins * delta_lambda
-        else:
-            raise AstronomicalObjectError("Error in rebinning Forest. "
-                                          "Class variable 'wave_solution' "
-                                          "must be either 'lin' or 'log'. "
-                                          f"Found: {Forest.wave_solution}")
+        delta_log_lambda = Forest.log_lambda_grid[1] - Forest.log_lambda_grid[0]
```

Comment:
When using linear pixels, delta_log_lambda will change with lambda. I guess this is ok, since this is only to have a buffer near the edge of our lambda_grid, but it would be good to add a comment in the code to clarify this.

Reply:
Agreed! But as you noticed further down, this does not seem limited to just having a buffer.

Reply:
this is a mistake from the merge conflicts. Here we should see the linear/logarithmic pixelisation to ensure that the chosen binning is kept. I'll restore this in a commit